src/app/GitExtUtils/GitUI/ComboBoxExtensions.cs

```diff
@@ -19,39 +18,102 @@ public static void AdjustWidthToFitContent(this ComboBox comboBox)
             }
         }
 
-        public static void ResizeDropDownWidth(this ComboBox comboBox, int minWidth, int maxWidth)
+        /// <summary>
+        ///  Resizes the drop-down width of a <see cref="ComboBox"/> to fit within provided bounds.
+        /// </summary>
+        /// <param name="comboBox">Target control</param>
+        /// <param name="minWidth">Minimum drop-down width</param>
+        /// <param name="maxWidth">Maximum drop-down width</param>
+        /// <param name="dpiScaleBounds">Whether to apply Dpi scaling to <paramref name="minWidth"/> and <paramref name="maxWidth"/>. Defaults to <c>true</c>.</param>
+        public static void ResizeDropDownWidth(this ComboBox comboBox, int minWidth, int maxWidth, bool dpiScaleBounds = true)
         {
-            if (comboBox is null)
+            ArgumentNullException.ThrowIfNull(comboBox);
+
+            if (dpiScaleBounds)
             {
-                throw new ArgumentNullException(nameof(comboBox));
+                minWidth = DpiUtil.Scale(minWidth);
+                maxWidth = DpiUtil.Scale(maxWidth);
             }
 
             int calculatedWidth = GetPreferredDropDownWidth(comboBox);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
 
-        public static void ResizeDropDownWidth(this ToolStripComboBox comboBox, int minWidth, int maxWidth)
+        /// <summary>
+        ///  Resizes the drop-down width of a <see cref="ToolStripComboBox"/> to fit within provided bounds.
+        /// </summary>
+        /// <param name="comboBox">Target control</param>
+        /// <param name="minWidth">Minimum drop-down width</param>
+        /// <param name="maxWidth">Maximum drop-down width</param>
+        /// <param name="dpiScaleBounds">Whether to apply Dpi scaling to <paramref name="minWidth"/> and <paramref name="maxWidth"/>. Defaults to <c>true</c>.</param>
+        public static void ResizeDropDownWidth(this ToolStripComboBox comboBox, int minWidth, int maxWidth, bool dpiScaleBounds = true)
         {
-            if (comboBox is null)
+            ArgumentNullException.ThrowIfNull(comboBox);
+
+            if (dpiScaleBounds)
             {
-                throw new ArgumentNullException(nameof(comboBox));
+                minWidth = DpiUtil.Scale(minWidth);
+                maxWidth = DpiUtil.Scale(maxWidth);
             }
 
-            int calculatedWidth = GetPreferredDropDownWidth(comboBox.Control);
+            int calculatedWidth = GetPreferredDropDownWidth((ComboBox)comboBox.Control);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
```

Comment:
This works because `toolStripComboBox.DropDownWidth = 123` actually redirects the setting of `DropDownWidth` to underlying wrapped `(ComboBox)comboBox.Control`.
Updated.